app/controllers/admin/secrets_controller.rb

```diff
@@ -37,21 +45,29 @@ def destroy
   private
 
   def secret_params
-    @secret_params ||= params.require(:secret).permit(:project_permalink, :key, :value)
+    @secret_params ||= params.require(:secret).permit(:project_permalink, :deploy_group_permalink, :environment_permalink, :key, :value)
   end
 
   def key
-    params[:id] || "#{secret_params.fetch(:project_permalink)}/#{secret_params.fetch(:key)}"
+    params[:id] || "#{secret_params.fetch(:environment_permalink)}/#{secret_params.fetch(:project_permalink)}/#{secret_params.fetch(:deploy_group_permalink)}/#{secret_params.fetch(:key)}"
```

Comment:
parse_secret_key knows how to read it, so the code to generate it should live there too to avoid breakage ...